env/work/srw_python/srwl_bl.py

```diff
@@ -802,21 +801,57 @@ def calc_all(self, _v, _op):
         if hasattr(_v, 'fdir'): self.dir_main = _v.fdir
 
         #---setup electron beam
-        if(hasattr(_v, 'ebm_nm')): #To improve
```

Comment:
This probably needs revision: removing / renaming 'ueb', moving parts of code to existing functions in srwl_bl.py and srwl_uti_src.py.